src/algo/mod.rs

```diff
@@ -618,12 +625,17 @@ impl<G> Iterator for MinSpanningTree<G>
         //  b. If the edge connects two disjoint trees in the pre-MST,
         //     add the edge.
         while let Some(MinScored(score, (a, b))) = self.sort_edges.pop() {
-            let g = self.graph;
             // check if the edge would connect two disjoint parts
-            if self.subgraphs.union(g.to_index(a), g.to_index(b)) {
+            let (a_index, b_index) = (g.to_index(a), g.to_index(b));
+            if self.subgraphs.union(a_index, b_index) {
+                let (&a_order, &b_order) = match (self.node_map.get(&a_index),
+                                                  self.node_map.get(&b_index)) {
+                    (Some(a_id), Some(b_id)) => (a_id, b_id),
+                    _ => panic!("Edge references unknown node"),
```

Comment:
I am not sure if `panic!` here is a good choice to be honest.

Reply:
What do you think would be a better option? That branch should only run if the graph or the iterator itself is in an inconsistent state, so it doesn't seem to be worth changing the return type to `Result`. I guess the other option would be to just `unwrap` the node indexes?

Reply:
Certainly open to handling it differently